src/python/director/ikplanner.py

```diff
@@ -1474,6 +1482,23 @@ def mergeWithDefaultIkParameters(self, ikParameters):
         newIkParameters.fillInWith(self.defaultIkParameters)
         return newIkParameters
 
+    def clipState(self, q, l, u):
+        state = np.array(q)
+        ret = np.clip(state, l, u)
+        if np.max(np.abs(state-ret)) > 1e-6:
+            print 'State is outside of joint limits! clipped to limits.'
+            return ret.tolist()
+        else:
+            return ret.tolist()
```

Comment:
one nitpick, would you mind updating this function for readability:

 - rename function and args to "clipJointPositions(self, jointPositions, lowerBound, upperBound):"
 - remove the else condition and un-indent the return statement.
 - don't create a new variable name, the first line could just be `jointPositions = np.array(jointPositions)`